pkg/scheduler/algorithm/predicates/utils.go

```diff
@@ -87,3 +94,67 @@ func SetPredicatesOrderingDuringTest(value []string) func() {
 		predicatesOrdering = origVal
 	}
 }
+
+// isCSIMigrationOn returns a boolean value indicating whether
+// the CSI migration has been enabled for a particular storage plugin.
+func isCSIMigrationOn(csiNode *storagev1beta1.CSINode, pluginName string) bool {
+	if csiNode == nil || len(pluginName) == 0 {
```

Comment:
So if driver is not installed on the node, we consider migration to be off on the node? Is that accurate - because this means that pods will still be scheduled on the node and volume operations will be handled by in-tree plugins but I thought that is not what we want to do.

Reply:
cc @davidz627

Reply:
> So if driver is not installed on the node, we consider migration to be off on the node

This is not accurate.

Determining whether migration is on for a plugin for Attach/Detach involves 2 things:
1) Migration is "on" on the master (feature flags)
2) Migration is "on" on the node which the volume is scheduled to (can be determined through the annotation "storage.alpha.kubernetes.io/migrated-plugins" on the CSINode API object for that node)

For all other operations determining whether migration is on for a plugin involves just:
1) Migration is "on" on the master/node wherever the operation is performed (can just be determined by looking at the feature flags)

Driver installed/not installed does not factor into the determination of migrated or not. If we determine a plugin is migrated with the above criteria, if the driver is not installed generally we throw an error mentioning that the driver was required but not installed.

Reply:
okay thanks for clarifying. I think `isCSIMigrationOn` check is okay. But still we aren't doing anything in the code here to prevent scheduling of pods for which driver is not installed (and migration is on). This should be fixed.